docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,164 @@
+---
+title: image-service-cloud-authentication
+authors:
+  - "@carbonin"
+creation-date: 2021-10-07
+last-updated: 2021-10-07
+---
+
+# Image Service Cloud Authentication
+
+## Summary
+
+Authentication between the image service and assisted service when deployed
+in the cloud requires a new strategy because the existing tokens used in the cloud
+are too large and too short lived to be included in a pre-signed URL. Instead
+of the existing authentication used in the cloud, a password and expiration will be
+generated and recorded in the infraEnv when it is updated and on demand through a new
+endpoint. This password will be used as authentication for requests to assisted service
+specifically for the endpoints required by the image service.
+
+## Motivation
+
+When the assisted service is deployed in the cloud, it relies on Red Hat
+SSO for user authentication and authentication tokens from Red Hat SSO are not
+usable with pre-signed URLs for two main reasons:
+
+* Size - these tokens are typically ~2000 characters long
+* Expiration - these tokens have a maximum lifespan of 15 minutes
+
+### Goals
+
+* Define an authentication strategy suitable for pre-signed URLs in the cloud. This includes a
+token with a longer expiration (several hours) and a reasonable size (10s to 100s of characters)
+which will be included directly in a download URL
+
+### Non-Goals
+
+* Tie an image download URL to a Red Hat account or organization
+* Maintain public/private key pairs to generate and issue tokens
+* Add additional external identity management components
+
+## Proposal
+
+The new authentication mechanism will be a simple secret and expiration date
+both of which are tied to an infraEnv. This secret will be included as a
+parameter in the image download URL and assisted service authentication will
+pass if the secret in the URL matches the one currently in the infraEnv record
+and the request is processed before the expiration time. A new secret will be
+generated any time a user updates the infraEnv in a way that would change the
+image content. Additionally a new endpoint will be required to force a new
+secret to be created in the event that the current one has expired and no new
+changes to the infraEnv are necessary.
```

Comment:
Why not have the service create a new one?

Reply:
I think a background job is more user-friendly, no?

Reply:
I suppose we already have one for expiring images. :man_shrugging: 

How would we decide when _not_ to renew the secret on a particular infra-env though?
In the cloud a user probably doesn't care about the infra env once they've installed the cluster, but the modeling doesn't make it super easy to determine when that's done. Or do we just keep regenerating the secret until the infra env gets removed?